packages/fx-core/src/plugins/resource/frontend/v3/ReactTabScaffoldPlugin.ts

```diff
@@ -0,0 +1,110 @@
+    return ok([
+      {
+        name: "ReactTab_JS",
+        language: "javascript",
+        description: "Tab frontend with React Framework using Javascript",
+      },
+      {
+        name: "ReactTab_TS",
+        language: "typescript",
+        description: "Tab frontend with React Framework using Typescript",
+      },
+    ]);
+  }
+  async scaffold(
+    ctx: v3.ContextWithManifest,
+    inputs: v3.PluginScaffoldInputs
+  ): Promise<Result<Json | undefined, FxError>> {
+    const solutionSettings = ctx.projectSetting.solutionSettings as v3.TeamsFxSolutionSettings;
+    ctx.logProvider.info(Messages.StartScaffold(this.name));
+    const progress = ctx.userInteraction.createProgressBar(
+      Messages.ScaffoldProgressTitle,
+      Object.entries(ScaffoldSteps).length
+    );
+    await progress.start(Messages.ProgressStart);
+    await progress.next(ScaffoldSteps.Scaffold);
+    const template = inputs.template;
+    const language = template === "ReactTab_TS" ? "ts" : "js";
+    const componentPath = path.join(inputs.projectPath, FrontendPathInfo.WorkingDir);
+    const hasFunction = solutionSettings.activeResourcePlugins.includes(
+      BuiltInResourcePluginNames.function
+    );
+    const variables = {
+      showFunction: hasFunction.toString(),
+    };
+    await scaffoldFromTemplates({
+      group: TemplateInfo.TemplateGroupName,
+      lang: language,
+      scenario: Scenario.Default,
+      templatesFolderName: FrontendPathInfo.TemplateFolderName,
+      dst: componentPath,
+      fileNameReplaceFn: removeTemplateExtReplaceFn,
+      fileDataReplaceFn: genTemplateRenderReplaceFn(variables),
+      onActionEnd: async (action: ScaffoldAction, context: ScaffoldContext) => {
+        if (action.name === ScaffoldActionName.FetchTemplatesUrlWithTag) {
+          ctx.logProvider.info(Messages.getTemplateFrom(context.zipUrl ?? Constants.EmptyString));
+        }
+      },
+      onActionError: async (action: ScaffoldAction, context: ScaffoldContext, error: Error) => {
+        ctx.logProvider.info(error.toString());
+        switch (action.name) {
+          case ScaffoldActionName.FetchTemplatesUrlWithTag:
+          case ScaffoldActionName.FetchTemplatesZipFromUrl:
+            // TelemetryHelper.sendScaffoldFallbackEvent(new TemplateManifestError(error.message));//TODO
+            ctx.logProvider.info(Messages.FailedFetchTemplate);
+            break;
+          case ScaffoldActionName.FetchTemplateZipFromLocal:
+            throw new TemplateZipFallbackError();
+          case ScaffoldActionName.Unzip:
+            throw new UnzipTemplateError();
+          default:
+            throw new UnknownScaffoldError();
+        }
+      },
+    });
+    await progress.end(true);
+    ctx.logProvider.info(Messages.EndScaffold(this.name));
+    const module = getModule(solutionSettings, inputs.module);
+    if (module) {
+      module.dir = FrontendPathInfo.WorkingDir;
+      module.deployType = "folder";
```

Comment:
Can I add customize config of module? For React tab, I may need to add module.indexPath = "/index.html#"

Reply:
And I remember that scaffold plugin can contribute to teams app manifest template, how to achieve that?